File: WarehouseManagementWeb.Application/Dto/Input/ResourceReceipt/UpdateResourceReceiptInput.cs
namespace WarehouseManagementWeb.Application.Dto.Input.ResourceReceipt
{
    /// <summary>
    /// Класс входной модели редактирования ресурса поступления.
    /// </summary>
    public class UpdateResourceReceiptInput
    {
        /// <summary>
        /// Id документа поступления.
        /// </summary>
        public int DocumentReceiptId { get; set; }

        /// <summary>
        /// Номер документа поступления. 
        /// </summary>
        public string? DocumentReceiptNumberCode { get; set; }

        /// <summary>
        /// Дата документа поступления.
        /// </summary>
        public DateTime Date { get; set; }

        /// <summary>
        /// Id клиента документа поступления.
        /// </summary>
        public int ClientId { get; set; }

        /// <summary>
        /// Список входящих ресурсов поступлений.
        /// </summary>
        public ICollection<ModifyResourceReceiptInput>? ModifyResourceReceiptInputs { get; set; }
    }
}

File: warehouse-management-web-angular/src/app/warehouse/document-receipt/models/input/update-resource-receipt-input.ts
import { ModifyResourceReceiptInput } from './modify-resource-receipt-input';

/**
 * Класс входной модели редактирования ресурса поступления.
 */
export class UpdateResourceReceiptInput {
  /**
   * Id документа поступления.
   */
  documentReceiptId: number = 0;

  /**
   * Номер документа поступления.
   */
  documentReceiptNumberCode: string = '';

  /**
   * Дата документа поступления.
   */
  date: Date = new Date();

  /**
   * Id клиента документа поступления.
   */
  documentReceiptClientId: number = 0;

  /**
   * Список входящих ресурсов поступлений.
   */
  modifyResourceReceiptInputs: ModifyResourceReceiptInput[] | null = null;
}
